source/mainnet/smart-contracts/tutorials/smartContractUpgrade/smartContractUpgrade.rst

```diff
@@ -0,0 +1,330 @@
+.. include:: ../../../variables.rst
+.. _smart-contract-upgrade:
+
+====================
+Native upgradability
+====================
+
+The goal of this tutorial is to deploy a ``contract-version1``, upgrade its logic, and migrate its state to ``contract-version2``.
+
+The `contract-version1 <https://github.com/Concordium/concordium-rust-smart-contracts/tree/main/examples/smart-contract-upgrade/contract-version1>`_
+includes an ``upgrade`` function and the `contract-version2 <https://github.com/Concordium/concordium-rust-smart-contracts/tree/main/examples/smart-contract-upgrade/contract-version2>`_
+includes a ``migration`` function. These functions are explored in the next two sections.
+
+The ``migration`` function
+==========================
+
+When deploying ``contract-version1``, you might not know the exact logic of the ``contract-version2`` yet.
+``Contract-versoin2`` might be a smart contract that fixes a security bug in your smart contract code which will be discovered at some point in the future.
```

Comment:
```suggestion
``contract-version2`` might be a smart contract that fixes a security bug in your smart contract code which will be discovered at some point in the future.
```

Reply:
I left the capital letter since it is the beginning of the sentence but fixed the typo in version.